base/abstractset.jl

```diff
@@ -229,14 +229,22 @@ end
 <( l::AbstractSet, r::AbstractSet) = l ⊊ r
 <=(l::AbstractSet, r::AbstractSet) = l ⊆ r
 
-function issubset(l, r)
-    if haslength(r)
-        rlen = length(r)
-        #This threshold was empirically determined by repeatedly
-        #sampling using these two methods (see #26198)
-        lenthresh = 70
+"""
+    hasfastin(T)
```

Comment:
Can we just make this `_hasfastin` (or make the docstring into a comment) in order to clarify that this is currently just an internal way of structuring the code.

Reply:
I would say this is more than just an internal way of structuring the code, it's an extensible trait. So I'm quite relunctant to write it `_hasfastin`. Internal methods don't have to start with an underscore. I also don't understand removing the docstring, as even with it this function is unexported and not documented in the manual. There is other code in base which is not public but with docstring (and without underscores).

Reply:
>  as even with it this function is unexported and not documented in the manual

Fair enough.

Reply:
IMO there's never a reason to not have a docstring for a function, public or not.  It's very helpful for exploring internals.